### PR TITLE
Fix minor inconsistencies and no-op sprintf detected by static analysis

### PR DESCRIPTION
**Pull request by TysonAndre**
and fix required param after optional param

| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | 
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by fabpot** [comment]
Are these changes only affecting 3.3? It looks like some of them apply to older versions (2.7 and 2.8).

**Comment by TysonAndre** [comment]
Created separate PRs for 2.7 and 2.8

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @TysonAndre.

File: src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php [hunk 1]
```diff
@@ -108,7 +108,7 @@ public function normalize($object, $format = null, array $context = array())
      *
      * @return string[]
      */
-    protected function getAttributes($object, $format = null, array $context)
+    protected function getAttributes($object, $format, array $context)
```
Comment (xabbuh) [inline src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php]: What is the reason for this change? This is breaking compatibility with classes that extend the `AbstractObjectNormalizer`.
